src/Acuminator/Acuminator.Analyzers/Resources.resx

```diff
@@ -285,9 +285,12 @@
   <data name="PX1042Title" xml:space="preserve">
     <value>In a RowSelecting handler, BQL statements and other database queries must be executed only inside a separate connection scope</value>
   </data>
-  <data name="PX1050Title" xml:space="preserve">
+  <data name="PX1050Title_HardcodedString" xml:space="preserve">
     <value>Hardcoded strings are not allowed as parameters for localization methods and PXException constructors. You should use string constants from the appropriate localizable messages class.</value>
   </data>
+  <data name="PX1050Title_NonConstFieldString" xml:space="preserve">
+    <value>Strings that are not constant fields are not allowed as parameters for localization methods and PXException constructors. You should use string constants from the appropriate localizable messages class.</value>
```

Comment:
@EAndrosova please review the new diagnostic message for the scenarios when string that is not a constant field is passed to the localization method or `PXException` constructor